scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/SFTPConnector.java

```diff
@@ -0,0 +1,378 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.examples;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import org.ow2.proactive.scheduler.common.task.TaskResult;
+import org.ow2.proactive.scheduler.common.task.executable.JavaExecutable;
+import org.zeroturnaround.zip.ZipUtil;
+
+import com.jcraft.jsch.Channel;
+import com.jcraft.jsch.ChannelSftp;
+import com.jcraft.jsch.JSch;
+import com.jcraft.jsch.JSchException;
+import com.jcraft.jsch.Session;
+import com.jcraft.jsch.SftpException;
+
+
+/**
+ * Import and export file(s) using SFTPConnector server.
+ *
+ * @author The ProActive Team
+ */
+public class SFTPConnector extends JavaExecutable {
+
+    private static final String DEFAULT_SFTP_HOSTNAME = "localhost";
+
+    private static final int DEFAULT_SFTP_PORT = 22;
+
+    private String sftpHostname = DEFAULT_SFTP_HOSTNAME;
+
+    private int sftpPort = DEFAULT_SFTP_PORT;
+
+    private boolean sftpExtractArchive;
+
+    private String sftpLocalRelativePath;
+
+    private String sftpRemoteRelativePath;
+
+    private String sftpMode;
+
+    private String sftpUsername;
+
+    private String sftpUrlKey;
+
+    private String sftpPassword;
+
+    private static final String SFTP_LOCAL_RELATIVE_PATH_ARG = "sftpLocalRelativePath";
+
+    @Override
+    public void init(Map<String, Serializable> args) throws Exception {
+        if (args.containsKey("sftpHostname")) {
+            sftpHostname = args.get("sftpHostname").toString();
+        }
+        if (args.containsKey("sftpPort")) {
+            sftpPort = Integer.parseInt(args.get("sftpPort").toString());
```

Comment:
You could catch a possible `NumberFormatException` here and throw a more explicit `IllegalArgumentException` that chains the `NumberFormatException`.

Reply:
Normally this should be prevented in the workflow by specifying the PA:Integer model for the port variable. But yes, that can't be always enough